lib/pure/net.nim

```diff
@@ -180,6 +188,22 @@ when defined(ssl):
     var errStr = ErrErrorString(err, nil)
     raise newException(SSLError, $errStr)
 
+  proc getSslContextExtraDataIndex*(): cint =
+    ## Retrieves unique index for storing extra data in SSLContext.
+    return SSL_CTX_get_ex_new_index(0, nil, nil, nil, nil)
```

Comment:
I don't think so, the index here is global for all SSLContexts. Something more fancy (generic generating global call to`getSslContextExtraDataIndex`?) might work.